[RM Ch03] Hardware Resources description added (#2144)
cntt-n/CNTT: doc/ref_model/chapters/chapter03.md

<a name="3.4"></a>

> ## 3.4 Hardware Infrastructure Resources

Reviewer: ```suggestion
## 3.4 Cloud Infrastructure Hardware Resources
```

Author: Chapter 3.1 points out the layering between the Virtualization Infrastructure Layers and the Hardware Virtualization Infrastructure Layer that both are part of the Cloud Infrastructure.
If we start making the very deep explanations of pre-pending everything with Cloud Infrastructure we will have to change in many places and it will not be very much clearer.

Reviewer: My comment is just about using consistent wording, not adding more explanation.
This chapter is not describing the Hardware Infrastructure Resources.

> Compute, Storage and Network resources serve as the foundation of the cloud infrastructure. They are exposed to and used by a set of networked Host Operating Systems in a cluster that normally handles the Virtualization Layer offering Virtual Machines or Containers where the application workloads (VNFs/CNFs) runs.

Reviewer 2: minor typo:
```suggestion
Compute, Storage and Network resources serve as the foundation of the cloud infrastructure. They are exposed to and used by a set of networked Host Operating Systems in a cluster that normally handles the Virtualization Layer offering Virtual Machines or Containers where the application workloads (VNFs/CNFs) run.
```

<p align="center"><img src="../figures/ch03-model-hardware-resources.png" alt="Cloud Infrastructure Hardware Resources" Title="Cloud Infrastructure Hardware Resources" width="65%"/></p>
<p align="center"><b>Figure 3-5:</b> Cloud Infrastructure Hardware Resources</p>

> In managed Hardware Infrastructure systems that can provision these consumed Compute, Storage and Network resources through operator commands or through software APIs there is a need to separate these offered resources, that shall be treated as leased resources, from the actual physical hardware resources that are installed in the datacenter. For that purpose, the hardware resource layer is split into a Logical Resource Layer that is visible for the software layer above and the Physical Resource Layer that is handled by the hardware installation and maintenance staff, and managed from the HW Infrastructure Management functions.

Reviewer 2: Suggest some simplification of the complex sentence:
```suggestion
In managed Hardware Infrastructure systems, these consumable Compute, Storage and Network resources can be provisioned through operator commands or through software APIs.  There is a need to distinguish between these consumable resources, that are treated as leased resources, from the actual physical hardware resources that are installed in the data center. For this purpose, the hardware resource layer is conceptually split into a Logical Resource Layer that surfaces the consumable resources to the software layer above, and the Physical Resource Layer that is operated and managed by the Data Center Operations team from the HW Infrastructure Management functions.
```

> In managed Hardware Infrastructure systems, these consumable Compute, Storage and Network resources can be provisioned through operator commands or through software APIs.  There is a need to distinguish between these consumable resources, that are treated as leased resources, from the actual physical hardware resources that are installed in the data center. For this purpose, the hardware resource layer is conceptually split into a Logical Resource Layer that surfaces the consumable resources to the software layer above, and the Physical Resource Layer that is operated and managed by the Data Center Operations team from the HW Infrastructure Management functions.

Reviewer: ```suggestion
In managed Cloud Infrastructure systems, these consumable Compute, Storage and Network resources can be provisioned through operator commands or through software APIs.  There is a need to distinguish between these consumable resources, that are treated as leased resources, from the actual physical hardware resources that are installed in the data center. For this purpose, the cloud infrastructure resource layer is conceptually split into a Logical Resource Layer that surfaces the consumable resources to the software layer above, and the Physical Resource Layer that is operated and managed by the Data Center Operations team using HW Infrastructure Management functions.
```

Author: I think we are blurring the management aspects of a clear separation between the Hardware Infrastructure resources as being Cloud Infrastructure Resources. Already now there are many types of resources that we have to manage.
The Hardware Infrastructure Layer has both (internal) Physical resources and the (externally exposed) consumable resources (Compute, Storage and Network that the Virtualization Infrastructure Layer are exposed to and by that consumes).
The Virtualization Infrastructure Layer instances has VM or Container with Network and Storage resources offered to their Tenants.

Reviewer: Same as above, I propose to use consistent wording, not more explanation.

Reviewer 3: @ulikleber, the issue here is that in this PR we are really talking about hardware and its management as the title itself says. Hence, replacing the word "hardware" by "infrastructure" is not appropriate in this context, as I think is what @TFredberg is saying.  I am not sure what is wrong in using the term "hardware" when we are talking about hardware :) Am I missing something?  How can we move on with this?

Reviewer: I am not concerned about using the term hardware. But the term "hardware infrastructure" also relates to equipment practice, power supply, cabinets and air-conditioning etc. Thus we are talking about the hardware of the cloud infrastructure, not about the management of fans and such. Therefore I suggested the term "cloud infrastructure **hardware**". It is not a big issue. If I am the only one concerned about "hardware infrastructure", let's keep it as is.

> Some installations might use a cluster of managed switches or storage components controlled by a Switch Fabric controller and/or a Storage Fabric controller acting as an appliance system. These systems should be federated with the HW Infrastructure Management system over some API to facilitate exchange of configuration intent, status and telemetry information allowing the HW Infrastructure Management and Management stack to automate Cloud Infrastructure operations. These systems normally also have their own Equipment Management APIs and procedures for the hardware installation and maintenance staff.

Reviewer 2: @TFredberg What does "...should be federated with the HW Infrastructure Management system over some API..." mean?

Author: The Cloud Infrastructure stack federated with a Switch Fabric means that:
- the CI shall be able to "send" configuration intent to the Switch Fabric
- the Switch Fabric shall be able  to "send" status and telemetry information to the CI
The word "send" is a very lose definition of getting a message across to the other side, and could be implemented in many different ways.
This allows HW Infrastructure Management and CI Management stack to have network automation that includes the switches taht are controlled by the federated Switch Fabric.
This would be a rather normal case for Opearators that have a separate Networking Department that owns and runs the Switch Fabric separately from the Data Center.

Reviewer 2: @TFredberg Thanks. Can these details be added?

Author: Added descriptive example of a "federated" Switch Fabric in Ch 3.4 and updated the branch.
